File: moos-ivp-labs/lab5.cpp
#include <iostream>
#include <string>
#include <sstream>
using namespace std;

class VertexSimple {
public:
    VertexSimple() {m_x=0; m_y=0;};
    VertexSimple(int x, int y) {m_x=x, m_y=y;};
    void setRandom(int min, int max) {
        int range = max - min;
        m_x = (rand() % range) + min;
        m_y = (rand() % range) + min;
    }
    string getSpec() {
        stringstream ss;
        ss << "x=" << m_x << ",y=" << m_y;
        return(ss.str());
    }
    
    void setXY(int x, int y) {m_x=x; m_y=y;};
    int  getX() const {return(m_x);};
    int  getY() const {return(m_y);};
protected:  
   int m_x;
   int m_y;
};

int main(){
    VertexSimple vertex;
    cout<<vertex.getSpec()<<endl;
    vertex.setRandom(0, 100);
    cout<<vertex.getSpec();
}

File: moos-ivp-labs/lab5.py
import random

class VertexSimple():
    def __init__(self, x = 0, y = 0):
        self._mx = x
        self._my = y
    def setRandom(self,min,max):
        self._mx = random.randint(min,max)
        self._my = random.randint(min,max)
    def getSpec(self):
        print(f'x={self._mx},y={self._my}')
    @property
    def X(self):
        return self._mx
    @property
    def Y(self):
        return self._my
    @X.setter
    def X(self,x):
        self._mx = x
    @Y.setter
    def Y(self,y):
        self._my = y

vertex = VertexSimple(5,5)
print(vertex.X, vertex.Y)
vertex.X = 10
vertex.Y = 10
vertex.getSpec()